frontend/src/components/dashboard/hooks/grouping.ts

```diff
@@ -0,0 +1,41 @@
+import { CurrentAnalysis } from "@/api/collimator/models/solutions/current-analysis";
+import { Category } from "../category";
+
+export interface CategorizedDataPoints {
+  category: Category;
+  dataPoints: DataPoint[];
+}
+
+export interface DataPoint {
+  groupName: string;
```

Comment:
isn't it a bit weird that the `groupName` would belong to `DataPoint`? Wouldn't that fit better under `AutomaticGroup`?
Also, what's the difference between `GroupBase.key` and `DataPoint.groupName`, in `AutomaticGroup`?

Reply:
Actually we want the group name (probably) being displayed as port of the tooltip when hovering meaning it makes sense to associate it with a datapoint. groupName and groupKey can be though of like 'id' and 'name'. The key is used internally and the name - two of which may or may not be the same - is what is shown to the user.

Alternatively we could also put the groupKey here + pass the list of groups to the analysis component which then puts the two together again to get the name? Or, of course, just drop it from the tooltip?